website/orders/api/v1/views.py

```diff
@@ -446,3 +480,35 @@ def patch(self, request, **kwargs):
             )
         else:
             return Response(status=status.HTTP_404_NOT_FOUND)
+
+
+class JoinShiftAPIView(APIView):
+    """Join Shift API View."""
+
+    serializer_class = ShiftSerializer
+    schema = CustomAutoSchema(response_schema={"$ref": "#/components/schemas/Shift"})
+    permission_required = "orders.can_manage_shift_in_venue"
+    permission_classes = [HasPermissionOnObject]
+
+    def get_permission_object(self):
+        """Get the object to check permissions for."""
+        obj = self.kwargs.get("shift")
+        return obj.venue
+
+    def patch(self, request, **kwargs):
+        """
+        Join Shift as baker view.
+
+        Permission required: orders.can_manage_shift_in_venue
+
+        This adds the requesting User to the Shift assignees.
+        """
+        shift = kwargs.get("shift")
```

Comment:
you could just add `shift` in the method signature here, so `def patch(self, request, shift, **kwargs):`

Reply:
Yes that is correct, but I did this in the entire codebase so let's fix this in one go sometime.

Reply:
Also: bit stupid of me but I didn't know otherwise.